common/test/acceptance/tests/video/test_video_module.py

```diff
@@ -226,13 +226,38 @@ def test_cc_button_wo_english_transcript(self):
         self.navigate_to_video()
         self.video.show_captions()
 
-        # Verify that we see "好 各位同学" text in the captions
+        # Verify that we see "好 各位同学" text in the transcript
         unicode_text = "好 各位同学".decode('utf-8')
         self.assertIn(unicode_text, self.video.captions_text)
 
-    def test_cc_button_transcripts_and_sub_fields_empty(self):
+    def test_cc_button(self):
+        """
+        Scenario: CC button works correctly with transcript in YouTube mode
+        Given the course has a video component in "Youtube" mode
+        And I have defined a transcript for the video
+        Then I see the closed captioning element over the video
         """
-        Scenario: CC button works correctly if transcripts and sub fields are empty,
+        data = {'transcripts': {'zh': 'chinese_transcripts.srt'}}
+        self.metadata = self.metadata_for_mode('youtube', data)
+        self.assets.append('chinese_transcripts.srt')
+        self.navigate_to_video()
+
+        # Show captions and make sure they're visible and cookie is set
+        self.video.show_closed_captions()
+        self.video.wait_for_closed_captions()
+        self.assertTrue(self.video.is_closed_captions_visible)
+        self.video.reload_page()
+        self.assertEqual(self.video.get_cookie('show_closed_captions')['value'], 'true')
```

Comment:
Checking the actual cookie probably does make more sense as a unit test, and this would test that the expected behavior is actually happening.  For this, it would just be a matter of replacing the assertEqual(get_cookie) with a wait for self.video.is_closed_captions_visible.